src/main/groovy/org/transmartproject/batch/concept/ConceptTree.groovy

```diff
@@ -40,6 +40,7 @@ class ConceptTree {
 
     void loadExisting(Collection<ConceptNode> nodes) {
         nodes.each { n ->
+            assert !n.new
```

Comment:
Maybe then `throw new IllegalArgumentException("Passed concept node deemed not new: $n")` would be more explicit without doing such assumptions about the calling context and implementation of `isNew`.

You're right that both solutions fails to provide the user with any useful information, but that's because the validation occurs too much downstream. Both `AssertionError` and `IllegalArgumentException` point to a programmatic error, where the issue here is database inconsistency. The proper place to notify the user of a missing `GatherCurrentConceptsTasklet` to present a proper message to the user is